bids/variables/io.py

```diff
@@ -85,6 +85,27 @@ def load_variables(layout, types=None, levels=None, skip_empty=True,
     return dataset
 
 
+def _get_timing_info(img_md):
+    if 'RepetitionTime' in img_md:
+        tr = img_md['RepetitionTime']
+        if 'DelayTime' in img_md:
+            ta = tr - img_md['DelayTime']
+        elif 'SliceTiming' in img_md:
+            slicetimes = sorted(img_md['SliceTiming'])
+            # a, b ... z
+            # z = final slice onset, b - a = slice duration
+            ta = np.round(slicetimes[-1] + slicetimes[1] - slicetimes[0], 3)
```

Comment:
That feels like a validation problem. While we can put those checks in, ad hoc, I think it would make sense to either fail on load for validation problems or be able to insert a little boilerplate check like: `validate(slicetimes, 'SliceTiming')`